### PR TITLE
Refactor ca menu

### PR DESCRIPTION
**Pull request by jmaslek**
No new commands.

Got rid of the l_also and l_similar inputs.  Adjusted the ability to `select` similar stocks or `add` specific stocks.  All commands just join ticker + similar (which is still set to 10 max).

Put historical plot on MinMaxScaler by default.  This way when looking at something worth 10 or 10k, it will overlap.

### CODE REVIEW

File: gamestonk_terminal/stocks/comparison_analysis/ca_controller.py [hunk 1]
```diff
@@ -89,61 +96,93 @@ def __init__(
...
+    scorr         sentiment correlation
+Finviz:
+    overview      brief overview comparison
+    valuation     brief valuation comparison
+    financial     brief financial comparison
+    ownership     brief ownership comparison
+    performance   brief performance comparison
+    technical     brief technical comparison
+
+    >po          portfolio optimization for selected tickers
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/comparison_analysis/ca_controller.py:141]: the > should be at the start of the line. And the portfolio should be aligned with above ones

**Review by DidierRLopes — COMMENTED** [review]
I got this:
```
(✨) (stocks)>(ca)> tsne
Wrong number of items passed 6, placement implies 1 
```
when loading NIO and using "get" at start.

Also the terminal crashed when I did:
```
(✨) (stocks)>(ca)> load tsla
Loading Daily TSLA stock with starting period 2020-08-31 for analysis.

(✨) (stocks)>(ca)> help
Traceback (most recent call last):
  File "terminal.py", line 282, in <module>
    terminal()
  File "terminal.py", line 255, in terminal
    process_input = t_controller.switch(an_input)
  File "terminal.py", line 122, in switch
    return getattr(
  File "terminal.py", line 161, in call_stocks
    return stocks_controller.menu()
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/stocks_controller.py", line 502, in menu
    process_input = stocks_controller.switch(an_input)
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/stocks_controller.py", line 174, in switch
    return getattr(
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/stocks_controller.py", line 345, in call_ca
    ret = ca_controller.menu(self.ticker, self.start, self.interval, self.stock)
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/comparison_analysis/ca_controller.py", line 938, in menu
    process_input = ca_controller.switch(an_input)
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/comparison_analysis/ca_controller.py", line 355, in switch
    return getattr(
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/comparison_analysis/ca_controller.py", line 361, in call_help
    self.print_help()
  File "/Users/DidierRodriguesLopes/Documents/git/GamestonkTerminal/gamestonk_terminal/stocks/comparison_analysis/ca_controller.py", line 99, in print_help
    s_intraday = (f"Intraday {self.interval}", "Daily")[self.interval == "1440min"]
TypeError: tuple indices must be integers or slices, not DataFrame
``

**Comment by DidierRLopes** [comment]
When 1 TESLA is loaded for instance:
```
(✨) (stocks)>(ca)> historical
No axis named 1 for object type Series 
```

**Comment by DidierRLopes** [comment]
```
(✨) (stocks)>(ca)> get
The limit of stocks to compare with are 10. Hence, 10 random similar stocks will be displayed.

[Finviz] Similar Companies: ALTR, AVPT, ETWO, HOOD, NETE, NLOK, PAYA, PAYO, SUMO, ZIXI 

(✨) (stocks)>(ca)> historical
'PLTR' 
```

Also seems broken, historical isn't outputting anything

**Comment by DidierRLopes** [comment]
> When 1 TESLA is loaded for instance:
> 
> ```
> (✨) (stocks)>(ca)> historical
> No axis named 1 for object type Series 
> ```

I think this is when there's a single stock loaded. We probably can do the "shadow commands" and add a check for a stock to compare. If it doesn't exist we report "load a stock to compare first" or something like that

**Comment by DidierRLopes** [comment]
If I load PLTR and then select AAPL, and use "hcorr", I get:

Captura de ecrã 2021-08-31, às 23 22 22

**Comment by jmaslek** [comment]
> If I load PLTR and then select AAPL, and use "hcorr", I get:
> 
> Captura de ecrã 2021-08-31, às 23 22 22

`load pltr`
`select aapl`
`hcorr`
Screen Shot 2021-09-01 at 9 32 53 AM

**Comment by jmaslek** [comment]
Fixed your comments except the one I cannot reproduce

**Comment by jmaslek** [comment]
> scorr also needs to have a ticker loaded to compare.
Yup
> * ">    po" is missing the spaces
Yup
> * I get this when trying tsne without any comparison ticker:
> 
> ```
> (✨) (stocks)>(ca)> tsne
> at least one array or dtype is required
> ```
So I don't replicate this.  As an example (I used SNDL - which is not in the sp500): 
```
stocks
load sndl
ca
tsne
```
I have no issues.  Even if I reload a new ticker

>  * my hcorr command completely disregards main ticker of interest
So it looks like the hcorr just alphabetizes everything.  Can you confirm it does all tickers, just not in the right order.  I should have the one of interest first so I will see if I can edit that.

> * Between github readme link and Daily stock there's 2 new lines instead of 1
Yup

**Review by DidierRLopes — APPROVED** [review]
Sweet. Thank you for bearing with all changes requests 😄

File: gamestonk_terminal/stocks/comparison_analysis/ca_controller.py [hunk 2]
```diff
@@ -89,61 +96,93 @@ def __init__(
...
+    q             quit this menu, and shows back to main menu
+    quit          quit to abandon program
+
+    load          load new base ticker
+    get           get similar companies
+    add           add more companies to current selected (max 10 total)
+    select        reset and select similar companies
+
+Custom Machine Learning Comparisons:
+    tsne          run TSNE on all SP500 stocks and returns 10 closest tickers
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/stocks/comparison_analysis/ca_controller.py:127]: So I thought I got rid of 'get' and put it into 4 separate commands?

File: gamestonk_terminal/stocks/comparison_analysis/ca_controller.py [hunk 3]
```diff
@@ -5,20 +5,26 @@
...
+from gamestonk_terminal.stocks.stocks_helper import load
+from gamestonk_terminal.stocks.comparison_analysis import polygon_model, finnhub_model
 from gamestonk_terminal.stocks.comparison_analysis import yahoo_finance_view
-from gamestonk_terminal.stocks.comparison_analysis import market_watch_view
+from gamestonk_terminal.stocks.comparison_analysis import marketwatch_view
 from gamestonk_terminal.stocks.comparison_analysis import finbrain_view
-from gamestonk_terminal.stocks.comparison_analysis import finviz_compare_view
+from gamestonk_terminal.stocks.comparison_analysis import (
+    finviz_compare_view,
+    finviz_compare_model,
```
[the remark this answers is not in the document]
Comment (jmaslek) [reply gamestonk_terminal/stocks/comparison_analysis/ca_controller.py:25]: So the _view functions shouldn't return anything.  The _model calls here return the list of similar which is assigned to self.  Unless I just do the api calls here in controller, I don't see how a model imported can be avoided.

**Pull request merged**